### PR TITLE
mungebot: Improve approval notification

### PR DESCRIPTION
**Pull request by apelisse**
Use golang templates rather than a string buffer to construct the
message as it's easier to read the logic and what the overall message
looks like. Also change some strings, and allow hiding/showing the
details of the message (especially when approved).

### CODE REVIEW

File: mungegithub/mungers/approvers/owners.go [hunk 1]
```diff
@@ -396,40 +396,57 @@ func (ap Approvers) IsApproved() bool {
...
-	return strings.Join(approvals, ", ")
+	return approvals
 }
 
 type File interface {
-	toString(string, string) string
+	String() string
 }
 
 type ApprovedFile struct {
```
Comment (grodrigues3) [inline mungegithub/mungers/approvers/owners.go]: Can you explain why are you are adding these two fields?

File: mungegithub/mungers/approvers/owners.go [hunk 2]
```diff
@@ -396,40 +396,57 @@ func (ap Approvers) IsApproved() bool {
...
-func (ua UnapprovedFile) toString(org, project string) string {
+func (ua UnapprovedFile) String() string {
 	fullOwnersPath := filepath.Join(ua.filepath, ownersFileName)
-	link := fmt.Sprintf("https://github.com/%s/%s/blob/master/%v", org, project, fullOwnersPath)
-	return fmt.Sprintf("- **[%s](%s)** \n", fullOwnersPath, link)
+	link := fmt.Sprintf("https://github.com/%s/%s/blob/master/%v", ua.org, ua.project, fullOwnersPath)
+	return fmt.Sprintf("- **[%s](%s)**\n", fullOwnersPath, link)
+}
+
+func GenerateTemplateOrFail(templ, name string, data interface{}) *string {
```
Comment (grodrigues3) [inline mungegithub/mungers/approvers/owners.go]: exported func missing comment

**Comment by grodrigues3** [comment]
Can you run the munger against some PRs in a test repo (mine is fine) and link to the results

File: mungegithub/mungers/approvers/owners.go (hunk 1 above)
Comment (apelisse) [inline mungegithub/mungers/approvers/owners.go]: I'm using `String() string` that is the idiomatic way in go to stringify an object. This makes it even more easy/obvious to call from the template.

File: mungegithub/mungers/approvers/owners.go (hunk 2 above)
Comment (apelisse) [inline mungegithub/mungers/approvers/owners.go]: Ack

**Comment by apelisse** [comment]
Sure: https://github.com/grodrigues3/contrib/pull/8#issuecomment-286547450. I think it looks much better.

**Pull request merged**